plugins/grow-smooth-generator/includes/CLI/AutomateWoo.php

```diff
@@ -0,0 +1,150 @@
+<?php
+namespace WC\Grow\SmoothGenerator\CLI;
+
+use WP_CLI, WP_CLI_Command;
+use WC\SmoothGenerator\Generator\Order;
+use WC\Grow\SmoothGenerator\Generator\AutomateWoo\Conversion;
+use WC\Grow\SmoothGenerator\Generator\AutomateWoo\EmailTracking;
+
+/**
+ * WP-CLI Integration class
+ */
+class AutomateWoo extends WP_CLI_Command {
+
+	/**
+	 * Generate conversions.
+	 *
+	 * @param array $args Arguments specified.
+	 * @param array $assoc_args Associative arguments specified.
+	 */
+	public static function conversions( $args, $assoc_args ) {
+		list( $amount ) = $args;
+		$amount         = absint( $amount );
+
+		$time_start = microtime( true );
+
+		WP_CLI::line( 'Initializing...' );
+
+		$progress = \WP_CLI\Utils\make_progress_bar( 'Generating conversions', $amount );
+
+		Order::disable_emails();
+		for ( $i = 1; $i <= $amount; $i++ ) {
+			Conversion::generate( true, $assoc_args );
+			$progress->tick();
+		}
+
+		$time_end       = microtime( true );
+		$execution_time = round( ( $time_end - $time_start ), 2 );
+		$display_time   = $execution_time < 60 ? $execution_time . ' seconds' : human_time_diff( $time_start, $time_end );
+
+		$progress->finish();
+
+		WP_CLI::success( $amount . ' conversions generated in ' . $display_time );
+	}
+
+	/**
+	 * Generate email tracking.
+	 *
+	 * @param array $args Arguments specified.
+	 * @param array $assoc_args Associative arguments specified.
+	 */
+	public static function email_tracking( $args, $assoc_args ) {
+		list( $amount ) = $args;
+		$amount         = absint( $amount );
+
+		$time_start = microtime( true );
+
+		WP_CLI::line( 'Initializing...' );
+
+		$progress = \WP_CLI\Utils\make_progress_bar( 'Generating email and SMS tracking', $amount );
+
+		for ( $i = 1; $i <= $amount; $i++ ) {
+			EmailTracking::generate( true, $assoc_args );
+			$progress->tick();
+		}
+
+		$time_end       = microtime( true );
+		$execution_time = round( ( $time_end - $time_start ), 2 );
+		$display_time   = $execution_time < 60 ? $execution_time . ' seconds' : human_time_diff( $time_start, $time_end );
+
+		$progress->finish();
+
+		WP_CLI::success( $amount . ' email and SMS tracking generated in ' . $display_time );
```

Comment:
This code is almost the same as `AutomateWoo::conversions`, and each time we add a command is going to be likely duplicated again. 

Maybe we can wrap this over something like `AutomateWoo::run( $generator, $args, $assoc_args )`